impl/src/main/java/org/jboss/weld/bean/DisposalMethod.java

```diff
@@ -123,7 +123,7 @@ private void checkDisposalMethod(EnhancedAnnotatedMethod<T, ? super X> enhancedA
         }
         if (declaringBean instanceof SessionBean<?>) {
             SessionBean<?> sessionBean = (SessionBean<?>) declaringBean;
-            Set<MethodSignature> businessMethodSignatures = sessionBean.getBusinessMethodSignatures();
+            Set<MethodSignature> businessMethodSignatures = sessionBean.getLocalBusinessMethodSignatures();
```

Comment:
Yes we have only: 
| If a non-static method of a session bean class has a parameter annotated @Disposes , and the
| method is not a business method of the session bean, the container automatically detects the
| problem and treats it as a definition error."